subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -185,14 +193,15 @@ static bool validate_signature(u32_t fw_src_address,
 		retval = rot_verify(fw_val_info->public_key,
 					key_data,
 					fw_val_info->signature,
-					(u8_t *)fw_src_address,
-					fwinfo->size);
+					(const u8_t *)fw_src_address,
+					fw_size);
 
 		if (retval == 0) {
 			for (u32_t i = 0; i < key_data_idx; i++) {
 				PRINT("Invalidating key %d.\n\r", i);
 				invalidate_public_key(i);
 			}
+			validated = true;
```

Comment:
Should we add an `else { validated = false }` here? I thing the zephyr code standard is to not allow initializing static variables to `false`, so it will look a bit unclear in that case without an explicit assignment to `false` here.